### PR TITLE
chap9

### PR DESCRIPTION
**Pull request by Syphung97**
ready

### CODE REVIEW

**Comment by Syphung97** [comment]
ready

File: .reek [hunk 1]
```diff
@@ -8,3 +8,5 @@ exclude_paths:
 "app/helpers":
   FeatureEnvy:
     enabled: false
+
```
Comment (DuongNT1512) [inline .reek]: xóa dòng thừa nhé

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -1,22 +1,32 @@
 class SessionsController < ApplicationController
   include SessionsHelper
-
+  attr_reader :user
   def new
```
Comment (DuongNT1512) [inline app/controllers/sessions_controller.rb]: cacsh 1 dòng so với trên  nhé

File: app/models/user.rb [hunk 3]
```diff
@@ -10,6 +11,35 @@ class User < ApplicationRecord
   before_save :email_down
   has_secure_password
 
+  class << self
+    def digest string
+      if ActiveModel::SecurePassword.min_cost
+        cost = BCrypt::Engine::MIN_COST
```
Comment (DuongNT1512) [inline app/models/user.rb]: cost = if ActiveModel::SecurePassword.min_cost
             BCrypt::Engine::MIN_COST
          else
             BCrypt::Engine.cost

File: app/models/user.rb [hunk 4]
```diff
@@ -10,6 +11,35 @@ class User < ApplicationRecord
...
+    end
+
+    def new_token
+      SecureRandom.urlsafe_base64
+    end
+  end
+
+  def remember
+    remember_token = User.new_token
+    update_attribute :remember_digest, User.digest(remember_token)
```
Comment (DuongNT1512) [inline app/models/user.rb]: convention
 update_attributes remember_digest: User.digest(remember_token)

File: app/views/sessions/new.html.erb [hunk 5]
```diff
@@ -11,6 +11,11 @@
       <%= f.label :password %>
       <%= f.password_field :password, class: "form-control" %>
 
+      <%= f.label :remember_me, class: "checkbox inline" do %>
+        <%= f.check_box :remember_me %>
+        <span>Remember me on this computer</span>
```
Comment (DuongNT1512) [inline app/views/sessions/new.html.erb]: i18n

File: app/views/users/new.html.erb [hunk 6]
```diff
@@ -19,7 +19,7 @@
       <%= f.password_field :password_confirmation, class: "form-control" %>
 
       <%= f.label :gender %>
-      <%= f.select :gender, @genders.keys %>
+      <%= f.select :gender, @genders.keys , :prompt =>  "Select"%>
```
Comment (DuongNT1512) [inline app/views/users/new.html.erb]: tifm hiểu nhé em https://apidock.com/rails/ActionView/Helpers/FormOptionsHelper/options_for_select

**Comment by Syphung97** [comment]
ready

**Comment by Syphung97** [comment]
ready

**Comment by Syphung97** [comment]
ready

**Comment by Syphung97** [comment]
ready

**Pull request merged**